cli/lsp/tsc.rs

```diff
@@ -89,46 +89,78 @@ impl TsServer {
 #[derive(Debug, Clone)]
 pub struct AssetDocument {
   pub text: String,
+  pub length: usize,
   pub line_index: LineIndex,
 }
 
+impl AssetDocument {
+  pub fn new<T: AsRef<str>>(text: T) -> Self {
+    Self {
+      text: text.as_ref().to_string(),
+      length: text.as_ref().encode_utf16().count(),
+      line_index: LineIndex::new(text),
+    }
+  }
+}
+
+#[derive(Debug, Clone)]
+pub struct Assets(HashMap<ModuleSpecifier, Option<AssetDocument>>);
+
+impl Default for Assets {
+  fn default() -> Self {
+    let assets = tsc::STATIC_ASSETS
+      .iter()
+      .map(|(k, v)| {
+        let url_str = format!("asset:///{}", k);
+        let specifier = ModuleSpecifier::resolve_url(&url_str).unwrap();
+        let asset = AssetDocument::new(v);
+        (specifier, Some(asset))
+      })
+      .collect();
+    Self(assets)
```

Comment:
This will load all libs into memory, regardless of if they are used. Can probably be optimized in the future to avoid some memory overhead.

Reply:
They only get allocated if the lsp gets used.  The alternative is to re-introduce some sort of `inner` for the assets and lazily calculate them when they are requested from tsc.